drivers/retained_mem/Kconfig.nrf

```diff
@@ -13,6 +13,7 @@ config RETAINED_MEM_NRF_RAM_CTRL
 	bool "nRF RAM retention driver"
 	default y
 	depends on DT_HAS_ZEPHYR_RETAINED_RAM_ENABLED && RETAINED_MEM_ZEPHYR_RAM && POWEROFF
-	depends on SOC_SERIES_NRF52X || SOC_SERIES_NRF53X || SOC_SERIES_NRF54LX || SOC_SERIES_NRF91X
+	depends on SOC_SERIES_NRF52X || SOC_SERIES_NRF53X || SOC_SERIES_NRF54LX || \
```

Comment:
isn't `depends on DT_HAS_ZEPHYR_RETAINED_RAM_ENABLE` enough and all those platform specific dependencies can be skipped?

Reply:
`DT_HAS_ZEPHYR_RETAINED_RAM_ENABLED` is `y` if there is at least one `retained-mem` region enabled in devicetree. However, underlying `nrfx_ram_ctrl` helper is not compatible with all devices, i.e. nRF51. This list of `depends on` explicitly list all of the devices supported by `nrfx_ram_ctrl` . Without it `nrfx_ram_ctrl` helper could be attempted to compile on unsupported device